edk2toolext/environment/repo_resolver.py

```diff
@@ -418,24 +418,27 @@ def clean(abs_file_system_path, ignore_files=[]):
         repo.git.clean(*params)
 
 
-def submodule_clean(abs_file_system_path, submodule, ignore_files=None):
+def submodule_clean(abs_file_system_path, submodule):
     """Resets and cleans a submodule of the repo.
 
     Args:
         abs_file_system_path (PathLike): repo directory
         submodule (obj): object containing path (relative) attribute
-        ignore_files (list, optional): list of files to ignore when performing a clean. Defaults to [].
 
     Raises:
         (GitCommandError): The command is invalid
         (InvalidGitRepositoryError): The repo is in an invalid format
         (NoSuchPathError): The path does not exist
         (ValueError): submodule's path was invalid
     """
-    submodule_path = Path(submodule.path).as_posix()
     with Repo(abs_file_system_path) as repo:
-        if repo.submodule(submodule_path).module_exists():
```

Comment:
`repo.submodule()` expects the submodule name, not path. For MU repos, the name and path are the same, but a submodule name does not have to be the relative path to the submodule, so we will find the submodule to update by comparing paths.